### PR TITLE
Fill in remaining C++ samples and tests that were blocked by lack of Box2D

### LINKED ISSUE #2786: `Rect2D`+`Box3D` migration to archetypes
Turn into sub-issues as needed.

- [x] Python support
  - [ ] Extensions
  - [x] Unit test (constructors, serialization)
  - [x] Doc example
- [x] Rust support
  - [x] Extensions
  - [x] Unit test (constructors, rust-to-rust roundtrip)
  - [x] Doc example
- [x] C++ support
  - [x] Extensions
  - [x] Unit test (constructors, serialization)
  - [x] Doc example
- [x] Cross-language roundtrip test
- [x] Remove old components
  - [x] Clean up `re_components`
  - [x] Port viewer to use new types and queries

### LINKED ISSUE #2787: `Linestrip{2|3}D` migration to archetypes
Turn into sub-issues as needed.

- [ ] Python support
  - [ ] Extensions
  - [ ] Unit test (constructors, serialization)
  - [ ] Doc example
- [ ] Rust support
  - [ ] Extensions
  - [ ] Unit test (constructors, rust-to-rust roundtrip)
  - [ ] Doc example
- [ ] C++ support
  - [ ] Extensions
  - [ ] Unit test (constructors, serialization)
  - [ ] Doc example
- [ ] Cross-language roundtrip test
- [ ] Remove old components
  - [ ] Clean up `re_components`
  - [ ] Port viewer to use new types and queries

### LINKED ISSUE #2789: `Point{2|3}D` migration to archetypes
Turn into sub-issues as needed.

- [ ] Python support
  - [ ] Extensions
  - [ ] Unit test (constructors, serialization)
  - [ ] Doc example
- [ ] Rust support
  - [ ] Extensions
  - [ ] Unit test (constructors, rust-to-rust roundtrip)
  - [ ] Doc example
- [ ] C++ support
  - [ ] Extensions
  - [ ] Unit test (constructors, serialization)
  - [ ] Doc example
- [ ] Cross-language roundtrip test
- [ ] Remove old components
  - [ ] Clean up `re_components`
  - [ ] Port viewer to use new types and queries

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes #2787 
* Fixes #2789
* Fixes #2786 
   * once  #3348 is in

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3351) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3351)
- [Docs preview](https://rerun.io/preview/a655edeb1a52100c77b9a5e6432b8eaaeab083f7/docs) 
- [Examples preview](https://rerun.io/preview/a655edeb1a52100c77b9a5e6432b8eaaeab083f7/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: rerun_cpp/src/rerun/datatypes/rotation3d_ext.cpp [hunk 1]
```diff
@@ -22,6 +22,10 @@ namespace rerun {
 #define Rotation3DExt Rotation3D
 #endif
 
-        const Rotation3DExt Rotation3DExt::IDENTITY = Rotation3DExt(Quaternion::IDENTITY);
+        // TODO(andreas): This constant initialization does not work for unknown reasons!
+        //                On clang(-mac) this set the Rotation3D::IDENTITY to all zero instead of
+        //                the expected quaternion.
+        // Using the same code as a non-constant works fine.
+        // const Rotation3DExt Rotation3DExt::IDENTITY = Quaternion::IDENTITY;
```
Comment (teh-cmc) [inline rerun_cpp/src/rerun/datatypes/rotation3d_ext.cpp:29]: oO

**Review by teh-cmc — APPROVED** [review]
Niceee!

File: rerun_cpp/src/rerun/datatypes/rotation3d_ext.cpp (hunk 1 above)
Comment (emilk) [reply rerun_cpp/src/rerun/datatypes/rotation3d_ext.cpp:29]: wth

Comment (Wumpf) [reply rerun_cpp/src/rerun/datatypes/rotation3d_ext.cpp:29]: I poked it for quite a while but couldn't figure it out, so I gave up for now :(

Comment (Wumpf) [reply rerun_cpp/src/rerun/datatypes/rotation3d_ext.cpp:29]: `Rotation` is a union type to I reviewed that again, but can't see what's wrong with it

**Pull request merged**